Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1029 @@
+  return polyline;
+}
+
+function createDefaultPath() {
+  var path = new PathGraphics();
+  path.leadTime = 0;
+  path.width = 4;
+  path.material = new PolylineOutlineMaterialProperty();
+  path.material.color = Color.RED;
+  path.material.outlineWidth = 2;
+  path.material.outlineColor = Color.BLACK;
+  return path;
+}
+
+// This is a list of the Optional Description Information:
+//  <cmt> GPS comment of the waypoint
+//  <desc> Descriptive description of the waypoint
+//  <src> Source of the waypoint data
+//  TODO <link> Link (URI/URL) associated with the waypoint
+//  <type> Type (category) of waypoint
+var descriptiveInfoTypes = {
+  time: {
+    text: "Time",
+    tag: "time",
+  },
+  comment: {
+    text: "Comment",
+    tag: "cmt",
+  },
+  description: {
+    text: "Description",
+    tag: "desc",
+  },
+  source: {
+    text: "Source",
+    tag: "src",
+  },
+  number: {
+    text: "GPS track/route number",
+    tag: "number",
+  },
+  type: {
+    text: "Type",
+    tag: "type",
+  },
+};
+
+var scratchDiv;
+if (typeof document !== "undefined") {
+  scratchDiv = document.createElement("div");
+}
+function processDescription(node, entity) {
+  var i;
+
+  var text = "";
+  var infoTypeNames = Object.keys(descriptiveInfoTypes);
+  var length = infoTypeNames.length;
+  for (i = 0; i < length; i++) {
+    var infoTypeName = infoTypeNames[i];
+    var infoType = descriptiveInfoTypes[infoTypeName];
+    infoType.value = defaultValue(
+      queryStringValue(node, infoType.tag, namespaces.gpx),
+      ""
+    );
+    if (defined(infoType.value) && infoType.value !== "") {
+      text = text + "<p>" + infoType.text + ": " + infoType.value + "</p>";
+    }
+  }
+
+  if (!defined(text) || text === "") {
+    //No description
+    return;
+  }
+
+  //Turns non-explicit links into clickable links.
+  text = autolinker.link(text);
+
+  //Use a temporary div to manipulate the links
+  //so that they open in a new window.
+  scratchDiv.innerHTML = text;
+  var links = scratchDiv.querySelectorAll("a");
+
+  for (i = 0; i < links.length; i++) {
+    links[i].setAttribute("target", "_blank");
+  }
+
+  var background = Color.WHITE;
+  var foreground = Color.BLACK;
+  var tmp = '<div class="cesium-infoBox-description-lighter" style="';
+  tmp += "overflow:auto;";
+  tmp += "word-wrap:break-word;";
+  tmp += "background-color:" + background.toCssColorString() + ";";
+  tmp += "color:" + foreground.toCssColorString() + ";";
+  tmp += '">';
+  tmp += scratchDiv.innerHTML + "</div>";
+  scratchDiv.innerHTML = "";
+
+  //return the final HTML as the description.
+  return tmp;
+}
+
+function processWpt(dataSource, geometryNode, entityCollection, sourceUri) {
+  var position = readCoordinateFromNode(geometryNode);
+  if (!defined(position)) {
+    throw new DeveloperError("Position Coordinates are required.");
+  }
+
+  var entity = getOrCreateEntity(geometryNode, entityCollection);
+  entity.position = position;
+  // TODO different icon support
```

Comment:
There's a couple of todos/commented out code here. Can you either implement or remove these? I think in this case, with different icon support, we can just remove this. GPX doesn't seem to have a way of specifying an icon for waypoints/tracks as far as I can tell. 

In general, it's OK if we don't have 100% of the spec implemented, ideally we can document what parts aren't documented if there are significant features we ignore, similar to the KMLDataSource https://cesium.com/docs/cesiumjs-ref-doc/KmlDataSource.html?classFilter=KMLDA. I don't think this is the case here.

Reply:
Removed